### ISSUE TITLE
Add window-local modes and global modes?

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#891]
Currently our modes are buffer-local.  What about window-local modes, or global modes?

For instance,  say we want to have a way to change all the paths of bookmarks, cookies, etc.  We could do this in the command, but this would not save any state and it would not allow us to revert the change.

With modes, we essentially create of a list of states which are all reversible.

Thoughts?

**Comment by Ambrevar** [comment]
See #874 for a discussion.

**Comment by jmercouris** [comment]
I think it will get confusing to have window-local modes. I would suggest buffer, and global.

**Comment by CorruptedVor** [comment]
> I think it will get confusing to have window-local modes. I would suggest buffer, and global.

I agree, but just in case someone would decide to have several windows for different purposes, I would say why hinder that use case? If you don't need it you don't use it.

**Comment by Ambrevar** [comment]
We've refactored our mode API dramatically for 3.0, so we will keep playing with what we have for a while until we get a better picture of how we can generalize it to windows. Something for 4.0 then!


For the time being, we have hooks and we can leverage those to automatically toggle modes on a per-window basis.
